src/librustdoc/clean/mod.rs

```diff
@@ -168,7 +168,7 @@ impl Clean<Type> for (ty::TraitRef<'_>, &[TypeBinding]) {
 
         debug!("ty::TraitRef\n  subst: {:?}\n", trait_ref.substs);
 
-        ResolvedPath { path, did: trait_ref.def_id, is_generic: false }
```

Comment:
This path is for a trait in a `ty::TraitRef` (e.g., a `where` clause trait bound), so its `Res` should be correct and preserve the `is_generic: false` behavior.